test/e2e/proxy_test.go

```diff
@@ -69,13 +68,48 @@ func testEnvVars(t *testing.T) {
 				t.Run(svcName, func(t *testing.T) {
 					svcEnvVars, err := tc.getProxyEnvVarsFromService(addr, svcName)
 					require.NoErrorf(t, err, "error getting environment variables of service %s", svcName)
-					for _, proxyVar := range supportedProxyVars {
+					for _, proxyVar := range watchedEnvVars {
 						assert.Equalf(t, expectedEnvVars[proxyVar], svcEnvVars[proxyVar], "incorrect value for %s", proxyVar)
 					}
 				})
 			}
 		})
 	}
+	var patches []*patch.JSONPatch
```

Comment:
flow looks good but could you add some comments to make it easier to understand? or move some logic into its own testEnvVarRemoval test method?